### DISCUSSION TITLE
How to modify the memory value of kong_healthchecks

### DISCUSSION THREAD

**Discussion by pzhx521** [Kong/kong#10094] [Help] [answered]
[nginx_kong tmpl link](https://github.com/Kong/kong/blob/3.1.1/kong/templates/nginx_kong.lua)
![image](https://user-images.githubusercontent.com/46491088/211707852-a8646da8-c77a-4425-99b5-e73f75c1f9fb.png)
As shown above, I want to modify the value of `lua_shared_dict kong_healthchecks           5m;` through parameters or environment variables, but I don't know how to do it.

I try to use environment variables or custom parameters, but their keys are` lua_ shared_ Dict`, so I can only set one successfully. The rest will be covered. As follows.
```sh
 export LUA_SHARED_DICT="kong_healthchecks 256m"
 export LUA_SHARED_DICT="kong_process_events 256m"
```
[init config code link](https://github.com/Kong/kong/blob/2f45e7ad3161e102f52ca4f70e361ae2851fad07/kong/conf_loader/init.lua#L1517)

**Comment by chronolaw** [comment, marked as answer]
We can use env `NGINX_HTTP_LUA_SHARED_DICT` to add a new shared dict, but we can not redefine the existing shared dict.

Some shared dicts (like `kong_healthchecks` and `kong_process_events`) are fixed in kong template, you can not change them unless you modify the template file.

**Reply by J1a-wei** [reply]
Hi @chronolaw . I got error logs recently. 
```
[error] 1097#0: *720540820 [lua] events.lua:273: post(): worker-events: failed posting event "healthy" by "lua-resty-healthcheck [0dc6f45b-8f8d-40d2-a504-473544ee190b:ethereum-mainnet-api.web3-helm.8599.svc]"; no memory

prometheus.lua:839: log_error(): Error while setting 'memory_workers_lua_vms_bytes{pid="1104",kong_subsystem="http"}' to '160569064': 'no memory'

log_error(): Error while setting 'upstream_target_health{upstream="heco-mainnet-api-graphql.web3-helm.8545.svc",target="[172.1.19.101:8545](http://172.1.19.101:8545/)",address="[172.1.19.101:8545](http://172.1.19.101:8545/)",state="dns_error",subsystem="http"}' to '0': 'no memory'
```

Kong Version: 2.7.0 (Helm chart)

Healthcheck has no memory, which will cause the service to be temporarily unavailable. Of course, it can be recovered soon. How to fix it? We run it in kubernetes. Discussion  https://github.com/Kong/kubernetes-ingress-controller/discussions/3338

**Reply by chronolaw** [reply]
If possible I suggest upgrading to kong 3.x, which introduced a new events library, and improved prometheus plugin.

**Reply by J1a-wei** [reply]
Hi @chronolaw , excuse me, an error occurred after I upgraded,  Cloud u help me ? Appreciate it! https://github.com/Kong/kong/discussions/10102
